### PR TITLE
bugfix/22679-tooltip-background-missing-with-usehtml

### LINKED ISSUE #22679: Tooltip background disappears when combining xAxis.events.setExtremes and tooltip.formatter with HTML
#### Expected behaviour

Tooltip will have its background at all times

#### Actual behaviour

Tooltip looses its background because the g.highcharts-tooltip element gets stuck with `opacity: 0; visibility: hidden`

#### Live demo with steps to reproduce
https://jsfiddle.net/h3m8qLev/15/

- Navigate a few points and we see that the tooltip has its background as expected
- Change the extremes a few times, either with the Range Selector buttons or the Navigator drag
- Go back to points. Now the tooltip content should render, but without the background

#### Product version
v12.1.2

#### Affected browser(s)
Chrome, Firefox

### PR DESCRIPTION
**Pull request by MarkusBarstad**
Fixed #22679 , tt background could be hidden by dragging rangeselector when using useHTML

### CODE REVIEW

File: ts/Core/Tooltip.ts [hunk 1]
```diff
@@ -1132,6 +1132,12 @@ class Tooltip {
 
             // Show it
             if (tooltip.isHidden && tooltip.label) {
+                // #22774
+                if (options.useHTML) {
+                    tooltip.label.element.style.opacity = 1 + '';
```
Comment (KacperMadej) [inline ts/Core/Tooltip.ts:1137]: Why not just `'1'`?

**Review by KacperMadej — CHANGES_REQUESTED** [review]
The code that runs after your addition should handle this correctly so it would be better to to make it work instead of directly modifying properties.

This will be merged after https://github.com/highcharts/highcharts/pull/22760 so please check for possible conflicts or create a different solution with that PR in mind.

**Comment by MarkusBarstad** [comment]
@KacperMadej Alright, I think I found a much better solution: clear the tooltip hide timer when rangeselector moves (the hide animation step function is the issue)

Do note that this involves modifying a stock file, just a headsup

**Comment by MarkusBarstad** [comment]
@KacperMadej I think I have solved it, the issue seemed to be that the tooltip `SVGElement` had a missmatch in its computed CSS property opacity  vs it's SVGElement opacity attribute.


Why they end up out of sync I have not yet been able to tell, but a fix can be to just make sure these are always in sync, as ive attempted to do here. Is this an OK solution? 🤔 

Note, the only failing tests are the parliament ones which fail on master as well currently.

**Review by KacperMadej — APPROVED** [review]
Works fine 👍
Seems sucha an edge case that the fix is enough IMO.
If you will struggle with creating a test you might try with something focused on the fix instead.

The only thing that is bugging me is the typing change - it is better to keep a more narrowed types with occasional casting instead of having `any` as a function argument.

Approving in advance.

**Review by TorsteinHonsi — APPROVED** [review]
